quic/codec/Decode.cpp

```diff
@@ -123,7 +123,8 @@ ReadAckFrame decodeAckFrame(
   uint64_t adjustedAckDelay = ackDelay->first << ackDelayExponentToUse;
```

Comment:
Alternatively can we declare adjustedAckDelay to be signed?

Reply:
Well, I'm not sure, I'm not really familiar with QUIC.

`ackDelay->first` is coming from `decodeQuicInteger()`, which looks like a decoder for variable-length integers.
8-byte integers are supported by this function, its return value is a `uint64_t`.

Left shifting this value by `ackDelayExponentToUse` (default is 3) can reach the sign bit of a signed 64-bit integer, and we don't want that; but I'm pretty sure I'm talking rubbish, because ack delays can't be that high.

Reply:
when a var-length int is 8 bytes, only 62 bits are actually used. But you are right, the shift afterwards can easily break it.
I will import this PR.